app/models/registration_repository.rb

```diff
@@ -61,7 +61,41 @@ def self.store_search_query(session, query)
     end
   end
 
-  # Gets stored search query params
+  # restore search query to render search screen again
+  def self.restore_search_query_options(session)
+    session = session.dup
```

Comment:
Is there a particular reason we don't want to call delete on the actual session object? In the interest in preserving past functionality, maybe keep all the session.delete calls on the original session object, but set the values on a new object that gets returned?